edxval/tests/test_api.py

```diff
@@ -1403,6 +1411,229 @@ def test_external_no_video_transcript(self):
+        """
+        # First create external transcript.
+        utils.create_file_in_fs(
+            transcript_data['file_data'],
+            transcript_file_name,
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR
+        )
+
+        # Verify that one video is present before import.
+        video_initial_count = 1
+        self.assertEqual(Video.objects.count(), video_initial_count)
+
+        # Verify that no transript was present before import.
+        self.assertEqual(VideoTranscript.objects.count(), 0)
+
+        # Import xml with empty edx video id.
+        api.import_from_xml(
+            etree.fromstring('<video_asset/>'),
+            '',
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR,
+            {
+                'en': [transcript_file_name]
+            }
+        )
+
+        # Verify that a new video is created.
+        videos = Video.objects.all()
+        self.assertEqual(len(videos), video_initial_count + 1)
+
+        # Verify that new transcript is created.
+        self.assertEqual(VideoTranscript.objects.count(), 1)
+
+        # Newly created video record.
+        edx_video_id = videos[1].edx_video_id
+
+        # Verify trancript record is created with correct data.
+        self.assert_transcripts(
+            edx_video_id,
+            [dict(transcript_data, video_id=edx_video_id)]
+        )
+
+    def test_external_multiple_video_transcript(self):
+        """
+        Verify that transcript import for external video working as expected when multiple transcripts are imported.
+        """
+        # First create external transcripts.
+        en_transcript_file_name = 'external-transcript-en.srt'
+        utils.create_file_in_fs(
+            constants.VIDEO_TRANSCRIPT_CUSTOM_SRT['file_data'],
+            en_transcript_file_name,
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR
+        )
+
+        es_transcript_file_name = 'external-transcript-es.srt'
+        utils.create_file_in_fs(
+            constants.VIDEO_TRANSCRIPT_CUSTOM_SRT['file_data'],
+            es_transcript_file_name,
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR
+        )
+
+        # Verify that one video is present before import.
+        video_initial_count = 1
+        self.assertEqual(Video.objects.count(), video_initial_count)
+
+        # Verify that no transript was present before import.
+        self.assertEqual(VideoTranscript.objects.count(), 0)
+
+        # Import xml with empty edx video id.
+        api.import_from_xml(
+            etree.fromstring('<video_asset/>'),
+            '',
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR,
+            {
+                'en': [en_transcript_file_name],
+                'es': [es_transcript_file_name]
+            }
+        )
+
+        # Verify that new video is created.
+        videos = Video.objects.all()
```

Comment:
1. No need to Query video via `edx_video_id`, it can be get from `api.import_from_xml`.
2. No need to Query Transcript since the assertion will be done in `self.assert_transcripts`.

Holds for all 4 tests